src/postgraphql/graphiql/graphiql.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+import { render } from 'react-dom'
+import GraphiQL from 'graphiql'
+import { buildClientSchema, introspectionQuery } from 'graphql'
+import 'graphiql/graphiql.css'
+
+class EnhancedGraphiQL extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      schema: undefined,
+      token: undefined,
+    }
+  }
+
+  componentWillMount() {
+    setupStream({
+      onChange: this.onChange,
+      onOpen: this.onOpen,
+      onError: this.onError,
+    })
+  }
+
+  render() {
+    return (
+      <GraphiQL schema={this.state.schema} fetcher={this.fetcher}>
+        <GraphiQL.Logo>PostGraphQL GraphiQL</GraphiQL.Logo>
```

Comment:
That’s really clever I like that 😄

However, I’m still opposed for more subjective reasons. As a user I’d probably just be annoyed. I think I’d be fine with that if PostGraphQL is being run as a CLI, but plain “GraphiQL” if it’s being run as a middleware.